### PR TITLE
feat: testing cassette recording order

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: src/modules/handler-executor.js [hunk 1]
```diff
@@ -12,8 +14,14 @@ module.exports = (options) => {
...
     execute: () => new Promise((resolve) => {
+      const cassetteFile = path.join(options.cassetteFolder, options.cassetteFile);
+      const hasCassette = fs.existsSync(cassetteFile);
+      // eslint-disable-next-line no-underscore-dangle
+      const pendingMocks = hasCassette ? nock.load(cassetteFile).map(e => e.interceptors[0]._key) : [];
+      const outOfOrderErrors = [];
+
       consoleRecorder.start();
-      nockBack.setMode('record');
+      nockBack.setMode(hasCassette ? 'lockdown' : 'record');
```
Comment (MacMcIrish) [inline src/modules/handler-executor.js]: Is this not what record mode means? Or was this a bug before?

Comment (simlu) [reply src/modules/handler-executor.js]: It does. Just making it explicit. No functionality change

File: test/mock/handler.js [hunk 2]
```diff
@@ -9,7 +9,20 @@ module.exports.returnUnix = (event, context, cb) => cb(null, { unix: Math.floor(
...
-  .get('http://ip-api.com/json', { json: true }, (err, res, body) => cb(err, body));
+  .get('http://ip-api.com/json', { json: true }).then(r => cb(null, r));
+module.exports.returnChainedExternal = async (event, context, cb) => {
+  const json = await request
+    .get('http://ip-api.com/json', { json: true });
+  const xmlCsv = await Promise.all([
+    request
+      .get('http://ip-api.com/xml', { json: true }),
+    request
+      .get('http://ip-api.com/csv', { json: true })
```
Comment (MacMcIrish) [inline test/mock/handler.js]: Odd formatting?

**Pull request merged**